[datadogmonitor] add getting started doc (#261)
DataDog/datadog-operator: docs/datadog_monitor.md

        ```

    1. Alternatively, update the [values.yaml][6] file of the Datadog Operator Helm chart to include your [Datadog API and application keys][5] and enable `DatadogMonitor`.
       Then, run

        ```shell
        helm install my-datadog-operator datadog/datadog-operator -f values.yaml
        ```

1. Create a file with the spec of your `DatadogMonitor` deployment configuration. A simple example configuration is:

    ```yaml
    apiVersion: datadoghq.com/v1alpha1
    kind: DatadogMonitor
    metadata:
    name: datadog-monitor-test
    spec:
      query: "avg(last_10m):avg:system.disk.in_use{*} by {host} > 0.5"
      type: "metric alert"
      name: "Test monitor made from DatadogMonitor"
      message: "We are running out of disk space!"
      tags:
        - "test:datadog"
    ```

    For additional examples, see [examples/datadog-monitor](../examples/datadogmonitor). Note that only metric alerts, query alerts, and service checks are supported.

1. Deploy the `DatadogMonitor` with the above configuration file:

    ```shell
    kubectl apply -f /path/to/your/datadog-monitor.yaml
    ```

>     If configured properly this results in the creation of a new monitor in your [Datadog account][7].

Reviewer: suggesting to change to
"Which will automatically create and update the monitor in Datadog. The monitor can be found in the [Manage Monitors] page in your account"

## Cleanup

> The following command deletes all the Kubernetes resources created by the above instructions:

Reviewer: should mention that the monitor will be removed from your Datadog account

```shell
kubectl delete datadogmonitor datadog-monitor-test
helm delete datadog
```

> ## Usage and Troubleshooting

Reviewer: it would be really valuable to show the outputs of these commands, as users would be interested to know which information they return and incentivize them to use them

To verify monitor creation and check the monitor state, run

```shell
kubectl get datadogmonitor datadog-monitor-test
```

To view details about the monitor, including monitor groups that are currently in an alerting state, run

```shell
kubectl describe datadogmonitor datadog-monitor-test
```

To investigate any issues, view the Operator logs (of the leader pod, if more than one):

```shell
kubectl logs <my-datadog-operator-pod-name>
```

[1]: https://helm.sh
[2]: https://kubernetes.io/docs/tasks/tools/install-kubectl/
[3]: https://github.com/DataDog/helm-charts/tree/master/charts/datadog-operator
[4]: https://artifacthub.io/packages/helm/datadog/datadog-operator
[5]: https://app.datadoghq.com/account/settings#api
[6]: https://github.com/DataDog/helm-charts/blob/master/charts/datadog-operator/values.yaml

> [7]: https://app.datadoghq.com/monitors/manage?q=tag%3A"generated%3Akubernetes"

Reviewer: should we mention that all DatadogMonitor CRDs will be tagged with "generated kubernetes"?

# Getting Started

This procedure describes the simplest and fastest way to deploy a `DatadogMonitor` with the Datadog Operator. **Note: Operator version 0.6+ is required.**

## Prerequisites

> Using the Datadog Operator requires the following prerequisites:

Reviewer: This is a bit confusing since it implies that these are the prerequisites for Datadog Operator instead of for DatadogMonitor CRD. Suggesting to change this line to "Install the Datadog Operator in your cluster using the following:"

Author: yea good catch, this was from a copy-paste. i'll fix

- **Kubernetes Cluster version >= v1.14.X**: Tests were done on versions >= `1.14.0`. However, it should work on versions `>= v1.11.0`. For earlier versions, due to limited CRD support, the Operator may not work as expected.
- [`Helm`][1] for deploying the `datadog-operator`.
- [`Kubectl` cli][2] for installing a `DatadogMonitor`.

## Adding a DatadogMonitor

To deploy a `DatadogMonitor` with the Datadog Operator, see the [`datadog-operator`][3] Helm chart.
Here are the steps:

> 1. Install the [Datadog Operator][4]:

Reviewer: Isn't that already covered in the Prerequisites? Feels like dup, maybe keep this and remove the above

# Getting Started

This procedure describes the simplest and fastest way to deploy a `DatadogMonitor` with the Datadog Operator.

## Prerequisites

Using the Datadog Operator requires the following prerequisites:

> - **Kubernetes Cluster version >= v1.14.X**: Tests were done on versions >= `1.14.0`. However, it should work on versions `>= v1.11.0`. For earlier versions, due to limited CRD support, the operator may not work as expected.

Reviewer 2: ```suggestion
- **Kubernetes Cluster version >= v1.14.X**: Tests were done on versions >= `1.14.0`. However, it should work on versions `>= v1.11.0`. For earlier versions, due to limited CRD support, the Operator may not work as expected.
```

- [`Helm`][1] for deploying the `datadog-operator`.
- [`Kubectl` cli][2] for installing a `DatadogMonitor`.

## Using DatadogMonitor

> To deploy a `DatadogMonitor` with the Datadog Operator, see the [`datadog-operator`][3] helm chart.

Reviewer 2: ```suggestion
To deploy a `DatadogMonitor` with the Datadog Operator, see the [`datadog-operator`][3] Helm chart.
```

Here are the steps:

> 1. Update the [values.yaml][4] file of the Datadog Operator helm chart to include your [Datadog API and application keys][5]

Reviewer 2: ```suggestion
1. Update the [values.yaml][4] file of the Datadog Operator Helm chart to include your [Datadog API and application keys][5]
```

1. Install the [Datadog Operator][6]:

   ```shell
   helm repo add datadog https://helm.datadoghq.com
   helm install my-datadog-operator datadog/datadog-operator -f values.yaml
   ```

> 

Reviewer 3: we could provide an alternative `helm install` command here as a single step (by passing the creds via the `--set` option)

Reviewer 3: note that the datadog monitor controller is [disabled by default](https://github.com/DataDog/datadog-operator/blob/master/main.go#L91) (at least in the next operator release). So we should make the necessary changes in helm chart and update this step accordingly

1. Create a file with the spec of your `DatadogMonitor` deployment configuration. A simple example configuration is:

    ```yaml
    apiVersion: datadoghq.com/v1alpha1
    kind: DatadogMonitor
    metadata:
    name: datadog-monitor-test

>     namespace: datadog

Reviewer 3: should we add a note about creating the namespace ?

Author: good catch; for simplicity i'll keep it in the default ns

    spec:
      query: "avg(last_10m):avg:system.disk.in_use{*} by {host} > 0.5"
      type: "metric alert"
      name: "Test monitor made from DatadogMonitor"
      message: "We are running out of disk space!"
      tags:
        - "test:datadog"
    ```

>     For additional examples, see [examples/datadog-monitor](../examples/datadogmonitor). Note that currently only metric alerts, query alerts, and service checks are supported.

Reviewer 2: ```suggestion
    For additional examples, see [examples/datadog-monitor](../examples/datadogmonitor). Note that only metric alerts, query alerts, and service checks are supported.
```

1. Deploy the `DatadogMonitor` with the above configuration file:
    ```shell
    kubectl apply -f /path/to/your/datadog-monitor.yaml
    ```

    If configured properly this results in the creation of a new monitor in your [Datadog account][7].

> 

Reviewer 3: Could be interesting to add a troubleshooting section, like checking the CR status or/and the operator logs

Author: good idea, thanks

>     If configured properly this will result in the creation of a new monitor in your [Datadog account][7].

Reviewer 2: ```suggestion
    If configured properly this results in the creation of a new monitor in your [Datadog account][7].
```
